lib/Installation/Partitioner/LibstorageNG/v4_3/FormatMountOptionsPage.pm

```diff
@@ -0,0 +1,131 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Handles page for common formatting and mounting options
+# when adding a partition using Expert Partitioner.
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+package Installation::Partitioner::LibstorageNG::v4_3::FormatMountOptionsPage;
+use strict;
+use warnings;
+
+sub new {
+    my ($class, $args) = @_;
+    my $self = bless {
+        app => $args->{app}
+    }, $class;
+    return $self->init($args);
+}
+
+sub init {
+    my $self = shift;
+    $self->{cb_filesystem}       = $self->{app}->combobox({id => '"Y2Partitioner::Widgets::BlkDeviceFilesystem"'});
+    $self->{cb_enable_snapshots} = $self->{app}->checkbox({id => '"Y2Partitioner::Widgets::Snapshots"'});
+    $self->{rb_format_device}    = $self->{app}->radiobutton({id => 'format_device'});
+    $self->{rb_no_format_device} = $self->{app}->radiobutton({id => 'no_format_device'});
+    $self->{rb_mount_device}     = $self->{app}->radiobutton({id => 'mount_device'});
+    $self->{cb_mount_point}      = $self->{app}->combobox({id => '"Y2Partitioner::Widgets::MountPoint"'});
+    $self->{rb_no_mount_device}  = $self->{app}->radiobutton({id => 'dont_mount_device'});
+    $self->{cb_encrypt}          = $self->{app}->checkbox({id => '"Y2Partitioner::Widgets::EncryptBlkDevice"'});
+    $self->{btn_next}            = $self->{app}->button({id => 'next'});
+    return $self;
+}
+
+sub enter_formatting_options {
+    my ($self, $formatting_options) = @_;
+    my $should_format    = $formatting_options->{should_format};
+    my $filesystem       = $formatting_options->{filesystem};
+    my $enable_snapshots = $formatting_options->{enable_snapshots};
+    $self->select_format()                if $should_format == 1;
+    $self->select_not_format()            if $should_format == 0;
+    $self->select_filesystem($filesystem) if $filesystem;
+    $self->enable_snapshots()             if $enable_snapshots;
+}
+
+sub select_format {
+    my ($self) = @_;
+    return $self->{rb_format_device}->select();
+}
+
+sub select_not_format {
+    my ($self) = @_;
+    return $self->{rb_no_format_device}->select();
+}
+
+sub select_filesystem {
+    my ($self, $filesystem) = @_;
+    my %filesystems = (
+        ext2  => 'Ext2',
+        ext3  => 'Ext3',
+        ext4  => 'Ext4',
+        btrfs => 'Btrfs',
+        fat   => 'FAT',
+        xfs   => 'XFS',
+        swap  => 'Swap',
+        udf   => 'UDF'
+    );
+    return $self->{cb_filesystem}->select($filesystems{$filesystem});
```

Comment:
I wonder if we want to check if `$filesystems{$filesystem}` is defined, as might complicate investigation in case of typo. Same goes to partition id. I've tried manually and REST API actually throws `Item: '' cannot be found` error. But error message here would make it more obvious. WDYT?

Reply:
I had the same doubt, but yes, why not, I add it for both an error.

Reply:
Awesome, thanks! Good job here overall ;)